packages/cattle-pushprox/templates/pushprox-clients.yaml

```
@@ -0,0 +1,159 @@
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      {{- if and .Values.clients.https.enabled .Values.clients.https.useServiceAccountCredentials }}
+      serviceAccountName: {{ template "pushProxy.client.name" . }}
+      {{- end }}
+      containers:
+      - name: pushprox-client
+        image: arvindiyengar/pushprox-linux-amd64:add_flag_for_token_path
+        command: [ "/app/pushprox-client" ]
+        args:
+        - --fqdn=$(HOST_IP)
+        - --proxy-url=$(PROXY_URL)
+        - --metrics-addr=$(PORT)
+        - --allow-port={{ required "Need .Values.metricsPort to configure client to be allowed to scrape metrics at port" .Values.metricsPort}}
+        {{- if .Values.clients.useLocalhost }}
+        - --use-localhost
+        {{- end }}
+        {{- if .Values.clients.https.enabled }}
+        {{- if .Values.clients.https.insecureSkipVerify }}
+        - --insecure-skip-verify
+        {{- end }}
+        {{- if .Values.clients.https.useServiceAccountCredentials }}
+        - --token-path=/var/run/secrets/kubernetes.io/serviceaccount/token
+        {{- end }}
+        {{- if .Values.clients.https.certDir }}
+        - --tls.cert=/etc/ssl/push-proxy/push-proxy.pem
+        - --tls.key=/etc/ssl/push-proxy/push-proxy-key.pem
+        - --tls.cacert=/etc/ssl/push-proxy/push-proxy-ca-cert.pem
+        {{- end }}
+        {{- end }}
+        env:
+        - name: HOST_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+        - name: PORT
+          value: :{{ .Values.clients.port }}
+        - name: PROXY_URL
+          value: {{ template "pushProxy.proxyUrl" . }}
+        {{- if and .Values.clients.https.enabled .Values.clients.https.certDir }}
+        volumeMounts:
+        - name: metrics-cert-dir
+          mountPath: /etc/ssl/push-proxy
+        {{- end }}
+        {{- if .Values.clients.resources }}
+        resources: {{ toYaml .Values.clients.resources | nindent 10 }}
+        {{- end }}
+      {{- if and .Values.clients.https.enabled .Values.clients.https.certDir }}
+      initContainers:
+      - name: copy-certs
+        image: alpine:latest
+        command:
+        - sh
+        - -c
+        - |
+          echo "Searching for files to copy within the source volume"
+          echo "cert: ${CERT_FILE_NAME}"
+          echo "key: ${KEY_FILE_NAME}"
+          echo "cacert: ${CACERT_FILE_NAME}"
+
+          CERT_FILE_SOURCE=$(find /etc/source/ -type f -name "${CERT_FILE_NAME}" | sort -r | head -n 1)
+          KEY_FILE_SOURCE=$(find /etc/source/ -type f -name "${KEY_FILE_NAME}" | sort -r | head -n 1)
+          CACERT_FILE_SOURCE=$(find /etc/source/ -type f -name "${CACERT_FILE_NAME}" | sort -r | head -n 1)
+          
+          test -z ${CERT_FILE_SOURCE} && echo "Failed to find cacert file" && exit 1
+          test -z ${CERT_FILE_SOURCE} && echo "Failed to find cert file" && exit 1
+          test -z ${CERT_FILE_SOURCE} && echo "Failed to find key file" && exit 1
```

These all do the same thing

---

Good catch, this should be:
```suggestion
          test -z ${CERT_FILE_SOURCE} && echo "Failed to find cacert file" && exit 1
          test -z ${KEY_FILE_SOURCE} && echo "Failed to find cert file" && exit 1
          test -z ${CACERT_FILE_SOURCE} && echo "Failed to find key file" && exit 1
```

Making the change in the next commit